esp_idf/spotflow/srcs/logging/spotflow_log_cbor.c

```diff
@@ -0,0 +1,235 @@
+
+#include "logging/spotflow_log_cbor.h"
+#include "cbor.h"
+
+/* optimized property keys */
+#define KEY_MESSAGE_TYPE 0x00
+#define LOGS_MESSAGE_TYPE 0x00
+#define KEY_BODY 0x01
+#define KEY_BODY_TEMPLATE 0x02
+/* (unused for now) */
+#define KEY_BODY_TEMPLATE_VALUES 0x03
+#define KEY_SEVERITY 0x04
+#define KEY_LABELS 0x05
+#define KEY_DEVICE_UPTIME_MS 0x06
+#define KEY_SEQUENCE_NUMBER 0x0D
+
+/**
+ * @brief To verify CBOR validity
+ * 
+ */
+#define CBOR_CHECK(a, str, goto_tag, ret_value, ...)                              \
+    do                                                                            \
+    {                                                                             \
+        if ((a) != CborNoError)                                                   \
+        {                                                                         \
+            SPOTFLOW_LOG("%s(%d): " str, __FUNCTION__, __LINE__, ##__VA_ARGS__); \
+            ret = ret_value;                                                      \
+            goto goto_tag;                                                        \
+        }                                                                         \
+    } while (0)
+
+static void indent(int nestingLevel)
+{
+    while (nestingLevel--) {
+        SPOTFLOW_LOG("  ");
+    }
+}
+
+static void dumpbytes(const uint8_t *buf, size_t len)
+{
+    while (len--) {
+        SPOTFLOW_LOG("%02X ", *buf++);
+    }
+}
+
+/**
+ * Decode CBOR data manually
+ */
+static CborError example_dump_cbor_buffer(CborValue *it, int nestingLevel)
```

Comment:
Is it necessary to have this function in the code? Wouldn't it be cleaner to just output the hex representation to the console and debug the CBOR format only if necessary and with tools outside the source code?